xbmc/PlayListPlayer.h

```diff
@@ -103,6 +103,11 @@ class CPlayListPlayer : public IMsgTargetCallback
    */
   int GetCurrentPlaylist() const;
 
+  /*! \brief Get currently active playlist or a default for current player
```

Comment:
> if I called this in this way GetCurrentPlaylist(PLAYLIST_NONE), what would it return? 

It depends, that's the point of the function :-)  When you call it like that, this function is supposed to answer the question "if I want to queue an item after the one being played right now where should I put it? Maybe in PLAYLIST_NONE?"

The problem with trac 15707 is that deciding where to place a queued item is not obvious. It depends on what media is being played, if any at all. It would be nicer if the answer only required a `if ... else` statement, but it doesn't.

> Why don't you move the logic directly to those places?

Because the logic is cumbersome, as you noticed. Keeping the logic in a function would make it easier to understand the code that uses it and prevent that future changes lead to inconsistent behavior.